### DISCUSSION TITLE
GrpcClient beans in multi-module project

### DISCUSSION THREAD

**Discussion by zZHorizonZz** [quarkusio/quarkus#31788] [Q&A] [answered]
I have a project which consists of two projects. Project A is a module containing shared stuff about grpc and generated code for grpc protos. Project B depends on project B. But unfortunately, I can't find a way to use beans generated in Project A in project B particularly GrpcClient which should be used as bean.

```java
@GrpcClient
ClientService service;
```
I have also tried using Instance as described here: [how-to-resolve-multi-module-classpath-beans-in-quarkus](https://stackoverflow.com/questions/65353220/how-to-resolve-multi-module-classpath-beans-in-quarkus), but this doesn't work either.

NOTE: I'm using gradle.

I tried using beans.xml and yandex plugin but no luck. I have found out that you can specify 
index via this:

```
quarkus.index-dependency.<name>.group-id=
quarkus.index-dependency.<name>.artifact-id=
```

But I don't know how to use this with multi-module project.

[a comment by an automated account is not shown]

**Comment by cescoffier** [comment, marked as answer]
The recommendation would be to copy the proto file in the various projects. What kind of project is ProjectA?

**Reply by zZHorizonZz** [reply]
I don't know if I understand the question "what kind of project" correctly, but I have basically multiple projects where there is quarkus graphql project which should then communicate via grpc with other services and both of these graphql and services have shared project (Project A) which contains proto files and generated code based on these proto files. I tried making it simple since in my previous project which was not based on quarkus I had the same approach and it work because I didn't use CDI.

**Reply by cescoffier** [reply]
So, ProjectA is not a Quarkus project. How does the proto generation work?

**Reply by zZHorizonZz** [reply]
Project A is quarkus project. The last sentence was just a example of why I'm trying to do it via a shared project which is quarkus, java-library. I'm currently new to using quarkus so accept my apology if I'm not using the right approach for this.

**Reply by zZHorizonZz** [reply]
So after some searching, I think I found the answer I was searching for. Basically what happened to me is this I was trying to run an integration test where I was trying to inject grpc client which doesn't work for @QuarkusIntegrationTest, and also which confused me I relayed too much on IDE warnings that said that there are no beans for grpc client so when I tried putting the same GrpcClient injection in the actual project it said the same thing, so I automatically came to the conclusion that it will not work and I didn't bother researching any further.
